src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -0,0 +1,363 @@
+            {
+                // Fail the orchestration with an error explaining that the function name is invalid.
+                string errorMessage = this.extension.GetInvalidOrchestratorFunctionMessage(functionName.Name);
+                dispatchContext.SetProperty(OrchestratorExecutionResult.ForFailure(errorMessage, details: null));
+                return;
+            }
+
+            ExecutionStartedEvent? startEvent = runtimeState.ExecutionStartedEvent;
+            if (startEvent == null)
+            {
+                // This should never happen, but it's almost certainly non-retriable if it does.
+                dispatchContext.SetProperty(OrchestratorExecutionResult.ForFailure(
+                    message: "ExecutionStartedEvent was missing from runtime state!",
+                    details: null));
+                return;
+            }
+
+            bool isReplaying = runtimeState.PastEvents.Any();
+
+            this.TraceHelper.FunctionStarting(
+                this.Options.HubName,
+                functionName.Name,
+                instance.InstanceId,
+                isReplaying ? "(replay)" : this.extension.GetIntputOutputTrace(startEvent.Input),
+                FunctionType.Orchestrator,
+                isReplaying);
+
+            // One-time logging/notifications for when the orchestration first starts.
+            if (!isReplaying)
+            {
+#if !FUNCTIONS_V1
+                DurableTaskExtension.TagActivityWithOrchestrationStatus(OrchestrationRuntimeStatus.Running, instance.InstanceId);
+#endif
+                await this.LifeCycleNotificationHelper.OrchestratorStartingAsync(
+                    this.Options.HubName,
+                    functionName.Name,
+                    instance.InstanceId,
+                    isReplay: false);
+            }
+
+            var context = new RemoteOrchestratorContext(runtimeState);
+
+            var input = new TriggeredFunctionData
+            {
+                TriggerValue = context,
+#pragma warning disable CS0618 // Type or member is obsolete (not intended for general public use)
+                InvokeHandler = async functionInvoker =>
+                {
+                    // Invoke the function and look for a return value. Trigger return values are an undocumented feature that we depend on.
+                    Task invokeTask = functionInvoker();
+                    if (invokeTask is not Task<object> invokeTaskWithResult)
+                    {
+                        // This should never happen
+                        throw new InvalidOperationException("The internal function invoker returned a task that does not support return values!");
+                    }
+
+                    // The return value is expected to be a JSON string of a well-known schema.
+                    string? triggerReturnValue = (await invokeTaskWithResult) as string;
+                    if (triggerReturnValue == null || triggerReturnValue[0] != '{')
```

Comment:
Is there a need to check for the opening `{` character, or was this done just out of abundance of caution? If we're trying to make sure the return value is a JSON, I would think that simply trying to parse would be better check of schema-compliance, right?